### PR TITLE
Fixed falsy path variable resolution

### PR DESCRIPTION
**Pull request by kunagpal**
With this change, non-string path variable values will not be considered for resolution.

### CODE REVIEW

File: test/unit/url.test.js [hunk 1]
```diff
@@ -392,6 +392,25 @@ describe('Url', function () {
...
+                    { id: 'alpha', value: 'get' },
+                    { id: 'beta', value: null },
+                    { id: 'gamma', value: NaN },
+                    { id: 'gamma', value: undefined },
+                    { id: 'epsilon', value: [] },
+                    { id: 'phi', value: {} }
+                ]
+            });
+
+            expect(url.getPath()).to.eql('/get');
```
Comment (czardoz) [inline test/unit/url.test.js]: The behavior should result in the path `/get/:beta/:gamma/:delta/:epsilon/:phi`

Comment (kunagpal) [inline test/unit/url.test.js]: Okay.

File: lib/collection/url.js [hunk 2]
```diff
@@ -240,13 +240,13 @@ _.assign(Url.prototype, /** @lends Url.prototype */ {
...
         var self = this,
-            segmentArray = (options && options.unresolved) ? this.path : _.map(this.path, function (segment) {
+            segmentArray = (options && options.unresolved) ? this.path : _.transform(this.path, function (res, segment) {
                 var value;
                 if (_.startsWith(segment, PATH_VARIABLE_IDENTIFIER)) {
                     value = self.variables.one(segment.slice(1));
                 }
-                return value ? value : segment;
-            }),
+                res.push(_.isString(value && value.value) ? value : segment);
```
Comment (czardoz) [inline lib/collection/url.js]:
```
_.isString(value.valueOf()) ? value.valueOf() : segment);
```

I would suggest renaming the variable `value` to `variable`.

Comment (kunagpal) [inline lib/collection/url.js]: Has been fixed.

**Pull request merged**